src/overlays/fbdemos/ovl_fbdemo_wipe3/z_fbdemo_wipe3.c

```diff
@@ -17,37 +19,159 @@ void TransitionWipe3_SetColor(void* thisx, u32 color);
+    TransitionWipe3_Init,     TransitionWipe3_Destroy,     TransitionWipe3_Update,
+    TransitionWipe3_Draw,     TransitionWipe3_Start,       TransitionWipe3_SetType,
+    TransitionWipe3_SetColor, TransitionWipe3_SetEnvColor, TransitionWipe3_IsDone,
 };
 
-#endif
+#define TRANS3_GET_SPEED(type) (type & 1)
+#define TRANS3_GET_TEX_INDEX(type) ((type >> 1) & 7)
+#define TRANS3_GET_COLOR(type) ((type >> 4) & 3)
+
+typedef enum {
+    /* 0 */ TRANS_WIPE3_DIR_IN,
+    /* 1 */ TRANS_WIPE3_DIR_OUT
+} TransitionWipe3Direction;
+
+typedef enum {
+    /* 0 */ TRANS_WIPE3_SPEED_SLOW,
+    /* 1 */ TRANS_WIPE3_SPEED_FAST
+} TransitionWipe3Speed;
+
+typedef enum {
+    /* 0 */ TRANS_WIPE3_COLOR_BLACK,
+    /* 1 */ TRANS_WIPE3_COLOR_WHITE,
+    /* 2 */ TRANS_WIPE3_COLOR_GRAY
+} TransitionWipe3Color;
+
+void TransitionWipe3_Start(void* thisx) {
+    TransitionWipe3* this = THIS;
+
+    this->isDone = false;
+    this->curTexture = sTransWipe3Textures[this->texIndex & 7];
+    if (this->speedType == TRANS_WIPE3_SPEED_SLOW) {
+        this->wipeSpeed = 20;
+    } else {
+        this->wipeSpeed = 10;
+    }
+    if (this->colorType == TRANS_WIPE3_COLOR_BLACK) {
+        this->color.rgba = RGBA8(0, 0, 0, 255);
+    } else if (this->colorType == TRANS_WIPE3_COLOR_WHITE) {
+        this->color.rgba = RGBA8(255, 255, 255, 255);
+    } else {
+        this->color.rgba = RGBA8(160, 160, 160, 255);
+    }
+    if (this->dir != TRANS_WIPE3_DIR_IN) {
+        this->scrollY = 0;
+    } else {
+        this->scrollY = 500;
+        if (this->texIndex == 2) {
+            func_8019F128(NA_SE_OC_SECRET_WARP_OUT);
+        }
+    }
+    guPerspective(&this->projection, &this->normal, 60.0f, 1.3333334f, 10.0f, 12800.0f, 1.0f);
+    guLookAt(&this->lookAt, 0.0f, 0.0f, 400.0f, 0.0f, 0.0f, 0.0f, 0.0f, 1.0f, 0.0f);
+}
+
+void* TransitionWipe3_Init(void* thisx) {
+    TransitionWipe3* this = THIS;
+
+    bzero(this, sizeof(TransitionWipe3));
+    return this;
+}
+
+void TransitionWipe3_Destroy(void* thisx) {
+}
+
+void TransitionWipe3_UpdateScrollY(void* thisx, f32 scroll) {
+    TransitionWipe3* this = THIS;
+
+    this->scrollY = (s32)(500.0f * scroll);
+}
+
+void TransitionWipe3_Update(void* thisx, s32 updateRate) {
+    TransitionWipe3* this = THIS;
+
+    if (this->dir != TRANS_WIPE3_DIR_IN) {
+        if ((this->scrollY == 0) && (this->texIndex == 2)) {
+            func_8019F128(NA_SE_OC_SECRET_WARP_IN);
+        }
+        this->scrollY += (this->wipeSpeed * 3) / updateRate;
+        if (this->scrollY >= 500) {
+            this->scrollY = 500;
+            this->isDone = true;
+        }
+    } else {
+        this->scrollY -= (this->wipeSpeed * 3) / updateRate;
+        if (this->scrollY <= 0) {
+            this->scrollY = 0;
+            this->isDone = true;
+        }
+    }
+}
+
+void TransitionWipe3_Draw(void* thisx, Gfx** gfxP) {
+    Gfx* gfx = *gfxP;
+    Mtx* modelView = &THIS->modelView[THIS->frame];
+    f32 scale = 14.8f;
+    Gfx* texScroll;
```

Comment:
```suggestion
void TransitionWipe3_Draw(void* thisx, Gfx** gfxP) {
    TransitionWipe3* this = THIS;
    Gfx* gfx = *gfxP;
    Mtx* modelView = &THIS->modelView[THIS->frame];
    f32 scale = 14.8f;
    Gfx* texScroll;
```
Using `this` doesn't work? Is there not enough space on the stack?

Reply:
I couldn't get it to work .Did make a scratch if anyone wants to try more. https://decomp.me/scratch/cNOUa
Discussion from a few days ago: https://discord.com/channels/688807550715560050/688851997352263820/1112408731376943154